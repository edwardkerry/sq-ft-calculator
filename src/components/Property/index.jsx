import React from 'react';
import { string, number } from 'prop-types';

const Property = ({address, price, floor_area}) => (
  <li className='property'>
    <p className='address'>Address: {address}</p>
    <p className='price'>Value: Â£{price}</p>
    <p className='floor_area'>Floor area(sq ft): {floor_area}</p>
  </li>
);

Property.propTypes = {
  address: string.isRequired,
  price: number. isRequired,
  floor_area: number.isRequired
};

export default Property;
