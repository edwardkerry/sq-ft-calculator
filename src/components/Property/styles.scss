li {
  margin: 5px;
  padding: 20px;
  border: 2px solid rgb(53, 58, 57);
  background-color: rgb(169, 170, 177);
}

.active {
  border: 2px solid rgb(65, 221, 73);
  background-color: rgb(113, 228, 113);
}
