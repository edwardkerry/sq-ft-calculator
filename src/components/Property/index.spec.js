import React from 'react';
import renderer from 'react-test-renderer';
import Property from './';

describe('Property', () => {
  const testAddress = "1 Test Street"
  const testPrice = 100000
  const testArea = 1000


  describe('when not selected', () => {
    it('renders ', () => {
      const component = renderer.create(
        <Property
          address={testAddress}
          price={testPrice}
          floorArea={testArea}
          selected={false}
          select={() => { }}

      />)
      const tree = component.toJSON()

      expect(tree).toMatchSnapshot()
    });

    const wrapper = shallow(<Property
      address={testAddress}
      price={testPrice}
      floorArea={testArea}
      selected={false}
      select={() => { }}
    />)

    it('displays the given address', () => {
      expect(wrapper.find('.address').text()).toEqual(`Address: ${testAddress}`)
    });

    it('displays the given price', () => {
      expect(wrapper.find('.price').text()).toEqual(`Value: Â£${testPrice}`)
    });

    it('displays the given floor area', () => {
      expect(wrapper.find('.floor_area').text()).toEqual(`Floor area(sq ft): ${testArea}`)
    });

    it('does not have an active className', () => {
      expect(wrapper.hasClass('active')).toEqual(false)
    })
  });

  describe('when selected', () => {
    it('renders ', () => {
      const component = renderer.create(
        <Property
          address={testAddress}
          price={testPrice}
          floorArea={testArea}
          selected={true}
          select={() => { }}

        />)
      const tree = component.toJSON()

      expect(tree).toMatchSnapshot()
    });

    const wrapper = shallow(<Property
      address={testAddress}
      price={testPrice}
      floorArea={testArea}
      selected={true}
      select={() => { }}
    />)

    it('has an active className', () => {
      expect(wrapper.hasClass('active')).toEqual(true)
    });
  });
});
