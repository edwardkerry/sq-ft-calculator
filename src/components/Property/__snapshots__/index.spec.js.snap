// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property when not selected renders  1`] = `
<li
  className="property"
  onClick={[Function]}
>
  <p
    className="address"
  >
    Address: 
    1 Test Street
  </p>
  <p
    className="price"
  >
    Value: £
    100000
  </p>
  <p
    className="floor_area"
  >
    Floor area(sq ft): 
    1000
  </p>
</li>
`;

exports[`Property when selected renders  1`] = `
<li
  className="property active"
  onClick={[Function]}
>
  <p
    className="address"
  >
    Address: 
    1 Test Street
  </p>
  <p
    className="price"
  >
    Value: £
    100000
  </p>
  <p
    className="floor_area"
  >
    Floor area(sq ft): 
    1000
  </p>
</li>
`;
